test/container_app_test.py

```diff
@@ -86,6 +86,95 @@ async def test_container_debug_snapshot(container_client, tmpdir, servicer):
             test_breakpoint.assert_called_once()
 
 
+@pytest.fixture(scope="function")
+def fake_torch_module():
+    module_path = os.path.join(os.getcwd(), "torch.py")
+    with open(module_path, "w") as f:
+        f.write(
+            """
+import dataclasses
+@dataclasses.dataclass
+class CUDA:
+    device_count = lambda self: 0
+    _device_count_nvml = lambda self: 2
+
+cuda = CUDA()
+"""
+        )
+
+    yield module_path
+    # Teardown: remove the torch.py file
+    os.remove(module_path)
+
+
+@pytest.fixture(scope="function")
+def weird_torch_module():
+    module_path = os.path.join(os.getcwd(), "torch.py")
+    with open(module_path, "w") as f:
+        f.write("IM_WEIRD = 42\n")
+
+    yield module_path
+
+    os.remove(module_path)  # Teardown: remove the torch.py file
```

Comment:
Will we leave this around if the test fails (or otherwise raises?)

Reply:
I thought about it but given it was just test execution I let it be. I guess it's annoying if it raises and you get a stray file on your devbox— i was thinking of CI execution and its ephemerality.

Reply:
Oh @mwaskom it should be fine https://docs.pytest.org/en/latest/how-to/fixtures.html#handling-errors-for-yield-fixture

Reply:
Oh nice, I didn't know about that.